src/components/linter.ts

```diff
@@ -1,248 +1,52 @@
 import * as vscode from 'vscode'
-import * as path from 'path'
-import * as fs from 'fs'
-import * as os from 'os'
-import {ChildProcessWithoutNullStreams, spawn} from 'child_process'
-import {EOL} from 'os'
 
 import type {Extension} from '../main'
+import { ChkTeX } from './linterlib/chktex'
+import { LaCheck } from './linterlib/lacheck'
 
 export class Linter {
-    private readonly extension: Extension
+    protected readonly extension: Extension
+    private readonly chktex: ChkTeX
+    private readonly lacheck: LaCheck
     private linterTimeout?: NodeJS.Timer
-    private readonly currentProcesses = Object.create(null) as { [linterId: string]: ChildProcessWithoutNullStreams }
 
     constructor(extension: Extension) {
         this.extension = extension
+        this.chktex = new ChkTeX(extension)
+        this.lacheck = new LaCheck(extension)
     }
 
-    private get rcPath() {
-        let rcPath: string
-        // 0. root file folder
-        const root = this.extension.manager.rootFile
-        if (root) {
-            rcPath = path.resolve(path.dirname(root), './.chktexrc')
+    private getLinters(scope?: vscode.ConfigurationScope | null): (ChkTeX | LaCheck)[] {
```

Comment:
We can change the type `vscode.ConfigurationScope | null` to `vscode.ConfigurationScope`.